filter_plugins/oo_filters.py

```diff
@@ -489,6 +492,78 @@ def to_padded_yaml(data, level=0, indent=2, **kw):
+            raise errors.AnsibleFilterError("|failed expects hostvars is a dict")
+
+        facts = {}
+        regex = re.compile('^openshift_.*')
+        for key in hostvars:
+            if regex.match(key):
+                facts[key] = hostvars[key]
+        return facts
+
+    @staticmethod
+    # pylint: disable=too-many-branches
+    def oo_persistent_volumes(hostvars, groups, persistent_volumes=None):
+        """ Generate list of persistent volumes based on oo_openshift_env
+            storage options set in host variables.
+        """
+        if persistent_volumes == None:
+            persistent_volumes = []
+        if 'hosted' in hostvars['openshift']:
+            hosted_vars = hostvars['openshift']['hosted']
+            # Registry storage
+            if 'registry_storage_type' in hosted_vars:
+                if hosted_vars['registry_storage_type'] != None:
+                    storage_type = hosted_vars['registry_storage_type']
+                    if storage_type not in ['nfs']:
+                        msg = "|failed invalid registry storage type: {0}".format(storage_type)
+                        raise errors.AnsibleFilterError(msg)
+                    # NFS
+                    if storage_type == 'nfs':
+                        if hosted_vars['registry_storage_host'] != None:
+                            storage_host = hosted_vars['registry_storage_host']
+                        elif len(groups['oo_nfs_to_config']) > 0:
+                            storage_host = groups['oo_nfs_to_config'][0]
+                        else:
+                            raise errors.AnsibleFilterError("|failed no storage host detected")
+                        volume_dir = hosted_vars['registry_storage_volume_dir']
+                        volume = hosted_vars['registry_storage_volume']
+                        storage_path = volume_dir + '/' + volume
+                        storage = {'nfs': {'server': storage_host, 'path': storage_path}}
+                    storage_capacity = hosted_vars['registry_storage_volume_capacity']
+                    persistent_volume = {'name': 'registry-volume',
+                                         'capacity': storage_capacity,
+                                         'access_modes': ['ReadWriteMany'],
+                                         'storage': storage}
+                    persistent_volumes.append(persistent_volume)
+        return persistent_volumes
+
+    @staticmethod
+    def oo_persistent_volume_claims(hostvars, persistent_volume_claims=None):
+        """ Generate list of persistent volume claims based on oo_openshift_env
+            storage options set in host variables.
+        """
+        if persistent_volume_claims == None:
+            persistent_volume_claims = []
+        if 'hosted' in hostvars['openshift']:
+            hosted_vars = hostvars['openshift']['hosted']
+            # Registry storage
+            if 'registry_storage_type' != None:
```

Comment:
How about we check `registry_storage_type in ["nfs","whatever"]` (things we know will have pv/pvc) and we can have some other variables associated with `openshift_hosted_registry_storage_type="s3"`.